katsdptelstate/rdb_reader.py

```diff
@@ -49,23 +50,35 @@ def end_sorted_set(self, key):
         self._zset = {}
 
 
-def load_from_file(client, filename):
+def load_from_file(callback, file):
     """Load keys from the specified RDB-compatible dump file.
 
     Parameters
-    ---------
-    client : :class:`redis.StrictRedis`-like
-        Redis client wrapper
-    filename : str
-        Filename of .rdb file to import
+    ----------
+    callback : :class:`rdbtools.RdbCallback`-like
+        Backend-specific callback that stores keys as RDB file is parsed
+    file : str or file-like object
+        Filename of .rdb file to import, or object representing contents of RDB
 
     Returns
     -------
     int
         Number of keys loaded
     """
-    callback = _Callback(client)
+    try:
+        size = os.path.getsize(file)
+    except TypeError:
+        start_pos = file.tell()
+        file.seek(0, 2)
+        size = file.tell() - start_pos
+        file.seek(start_pos)
+    except AttributeError:
+        size = -1   # File object does not support seeking
+    logger.debug("Loading data from RDB dump of %s bytes",
+                 str(size) if size >= 0 else 'unknown')
     parser = RdbParser(callback)
-    logger.debug("Loading data from RDB dump of %d bytes", os.path.getsize(filename))
-    parser.parse(filename)
+    try:
+        parser.parse(file)
+    except TypeError:
```

Comment:
Are you guaranteed that `parser.parse` will always raise a `TypeError`, rather than in future raising something like `AttributeError`? I think it might be better to explicitly decide in this function whether you're dealing with a filename or a file object.

Reply:
Now do an explicit `open` and always use `parse_fd`.